src/banman.cpp

```diff
@@ -72,27 +72,26 @@ int BanMan::IsBannedLevel(CNetAddr net_addr)
 {
     // Returns the most severe level of banning that applies to this address.
     // 0 - Not banned
-    // 1 - Automatic misbehavior ban
+    // 1 - Discouraged because of misbehavior
     // 2 - Any other ban
-    int level = 0;
     auto current_time = GetTime();
     LOCK(m_cs_banned);
     for (const auto& it : m_banned) {
         CSubNet sub_net = it.first;
         CBanEntry ban_entry = it.second;
 
         if (current_time < ban_entry.nBanUntil && sub_net.Match(net_addr)) {
-            if (ban_entry.banReason != BanReasonNodeMisbehaving) return 2;
-            level = 1;
+            return 2;
         }
     }
-    return level;
+    return m_discouraged.contains(net_addr.GetAddrBytes());
```

Comment:
it feels a bit odd to treat what is essentially an enum (0/1/2) as an int like this, suggest replacing with a conditional returning 1/0 explicitly

Reply:
Ok, added an explicit `? 1 : 0`.